resources/google_drive_upload/drive.py

```diff
@@ -0,0 +1,114 @@
+import httplib2
+import os
+from dotmap import DotMap
+
+from apiclient import discovery, errors
+from apiclient.http import MediaFileUpload
+from oauth2client import client
+from oauth2client import tools
+from oauth2client.file import Storage
+
+# If modifying these scopes, delete your previously saved credentials
+# at ~/.credentials/ml_template_gdrive.json
+SCOPES = "https://www.googleapis.com/auth/drive.file"
+APPLICATION_NAME = "Drive API Python Quickstart"
+DEBUG = False
+
+
+def get_credentials():
```

Comment:
I am not sure I understand this function... Is it possible to specify user&pass instead of the secret file? It doesn't look like that, but then I have no idea what is does....

Reply:
The function is essentially copied from a Drive API example. It gets credentials for this "app" using OAuth2. So, when running this script for the first time, it will notice that `~/.credentials/ml_template_gdrive.json` doesn't exist, and will print a link, where you can auth to your Google account through your browser, and then will save the credentials to the above-mentioned path for future runs.

Reply:
Ok, I saw how it works in the CIL notebook. Seems legit.